jason/jason-dummy/src/utils/arrays.rs

```diff
@@ -1,22 +1,34 @@
+//! Functionality for passing arrays from Rust to Dart.
+
 use std::{marker::PhantomData, slice};
 
+use crate::ForeignClass;
+
+/// Array of pointers to [`ForeignClass`] structs.
+///
+/// Can be safely returned from extern functions. Foreign code must manually
+/// free this array by calling [`PtrArray_free`].
 #[repr(C)]
 pub struct PtrArray<T = ()> {
+    /// Array length.
     len: u64,
+    /// Pointer to the first element.
     arr: *const *mut (),
```

Comment:
@alexlapa shouldn't we use [`c_void` here](https://doc.rust-lang.org/std/ffi/enum.c_void.html)? It seems that `*mut ()` isn't a strict equivalent of `c_void`. Plus, [pointing to ZSTs in Rust has its own tricks](https://doc.rust-lang.org/nomicon/exotic-sizes.html#zero-sized-types-zsts).

Reply:
@tyranron ,

Номикон про это вот что говорит:

> *const () (or equivalent) works reasonably well for void*, and can be made into a reference without any safety problems. It still doesn't prevent you from trying to read or write values, but at least it compiles to a no-op instead of UB.

Ну *const () много где используют, например: [tokio](https://github.com/tokio-rs/tokio/blob/b42f21ec3e212ace25331d0c13889a45769e6006/tokio/src/macros/scoped_tls.rs#L26), [crossbeam](https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-utils/src/sync/parker.rs#L179), [bytes](https://github.com/tokio-rs/bytes/blob/master/src/bytes.rs#L987), [futures](https://github.com/rust-lang/futures-rs/blob/master/futures-task/src/waker.rs#L34).

Так что, вроде как, норм. Но не столь принципиально, на самом деле, могу поменять на `c_void`, если Вы считаете что так будет корректнее.

Reply:
@alexlapa let's change it. It will be more verbose in sematic terms, which is good.